### ISSUE TITLE
PPO2 with MlpLstmPolicy crashes GPU

### ISSUE THREAD

**Issue by SerialIterator** [hill-a/stable-baselines#172]
**Describe the bug**
While training using PPO2 with MlpLstmPolicy on custom env, my computer intermittently freezes yet continues training. When I attempt to monitor GPU's with `watch -n0.5 nvidia-smi` it loads first gpu data then seems to hang for a while until I see that my second GPU has an error. Even after training, anything that uses a gpu glitches requiring me to reset the computer even to run another model. I've run the same training on the same env using just MlpPolicy and it trains just fine (although my problem needs a recurrent network so I always get bad results) and I can monitor everything without GPU glitches. I thought it might be memory overload but I don't get anywhere near using all the ram or GPU memory.

**Code example**
```
import gym
import money_maker
import os

from stable_baselines.common.policies import MlpLstmPolicy, LstmPolicy
from stable_baselines.common.vec_env import SubprocVecEnv
from stable_baselines import PPO2

# multiprocess environment
n_cpu = 128
env = SubprocVecEnv([lambda: gym.make('maker-v0') for i in range(n_cpu)])

model = PPO2(MlpLstmPolicy, env, verbose=1, nminibatches=32,
             tensorboard_log="./ppo2_lstm_21_jan_morn_tensorboard/")

model.learn(total_timesteps=10000000)
model.save("ppo2_maker_lstm")
del model
env.close()
```

**System Info**
Describe the characteristic of your environment:
 * installed using PIP according to docs
 * 2- gtx-1080ti GPU's, driver:410.93, 
 * python version 3.6.5 in conda environment
 * Tensorflow version: 1.12.0

**Additional context**
![image](https://user-images.githubusercontent.com/33410869/51511117-d8cad100-1db4-11e9-919a-94465e6a8ee0.png)

**Comment by SerialIterator** [comment]
Hi araffin,
As I was trying to get a recurrent model to work I set `n_cpu=4` and `n_minibatches=2`as all I saw in the code as a requirement was `n_cpu` needed to be a multiple of `n_minibatches` so I started as small as possible to test and that is when I first noticed the GPU error. I didn't see much memory used and the cpu cores and gpu both were less than 50% utilized. I didn't think switching to lstm could cause the gpu fault so I kept resetting and increasing `n_cpu` and `n_minibatches` each time just to see if I could speed up training.

The `observation_space` is a 1D tensor with 21 features [21,]. I set them all to `np.finfo(np.float32).max` as their values can vary a lot, and my GPU's only want float32, but mainly because that's what I saw in CartPole to account for infinite values (lazy yes). I pass them as low (-high) and high to 
`spaces.Box(-high, high, dtype=np.float32)`. 

```
high = np.array([np.finfo(np.float32).max, np.finfo(np.float32).max,
                 np.finfo(np.float32).max, np.finfo(np.float32).max,
                 np.finfo(np.float32).max, np.finfo(np.float32).max,
                 np.finfo(np.float32).max, np.finfo(np.float32).max,
                 np.finfo(np.float32).max, np.finfo(np.float32).max,
                 np.finfo(np.float32).max, np.finfo(np.float32).max,
                 np.finfo(np.float32).max, np.finfo(np.float32).max,
                 np.finfo(np.float32).max, np.finfo(np.float32).max,
                 np.finfo(np.float32).max, np.finfo(np.float32).max,
                 np.finfo(np.float32).max, np.finfo(np.float32).max,
                 np.finfo(np.float32).max])

self.action_space = spaces.Discrete(3)
self.observation_space = spaces.Box(-high, high, dtype=np.float32)
```

I feed the data from a CSV file and figured I'd need to normalize the data first (which I haven't yet). I figured normalizing would help with training accuracy and speed but not memory or GPU issues. Values can be from 0.0 to around 3000.00 in the observations.

I ran training just now with `n_cpu=8` and `n_minibatches=2` and I'm getting the same GPU problem again. The model is training at around 1900 fps but anytime I try to view GPU status using `nvidia-smi` that terminal will freeze and my mouse becomes unresponsive for quite some time....

I just checked the tensorboard_log size and just the 3 logs I've saved were 40GB. It's being saved on my root filesystem and it's almost full. I think the code is trying to save more log data than I have space right now. I'm gonna free up some space and change the save location as that seems more like the culprit than a recurrent model. I am really surprised how much data tensorboard logs use and how much memory tensorboard uses. Thanks for replying, I'll free up some space then report back.

**Comment by araffin** [comment]
>I just checked the tensorboard_log size and just the 3 logs I've saved were 40GB. 

Yes, I agree that is too much. I'll be working on that in the coming weeks on the `enhancements` branch.

**Comment by SerialIterator** [comment]
I cleared up space on my root partition and ran the code again without tensorboard logging and had no problems. No freezing, I could monitor GPU's and no GPU error so it seems it was all due to low memory on my root partition. I haven't ran the code with tensorboard logging enabled again though. I looked for any parameters I could pass `TensorboardWriter` to only include info such as reward, loss, and learning rate, in order to reduce the file size, but only found the file `stable_baselines.common.base_class` which has the `TensorboardWriter` class. I didn't see any way of specifying what data it should write to logs. Is there an easy way to limit the info saved to logs?

**Comment by SerialIterator** [comment]
I tried using tensorboard logs with `MlpLnLstmPolicy` with `n_cpu = 8` and `nminibatches = 2` and the log (singular) grew to 51GB before hitting a million timesteps as well as crashing my second GPU (even though I'm running the model on my main GPU). Definitely has something to do with Tensorboard logging at this point.

**Comment by araffin** [comment]
> Is there an easy way to limit the info saved to logs?

For now, you need to have a local copy of stable-baselines and comment out the unnecessary tensorboard logging. In the near future, the minimal logging will be the default one.

**Comment by hill-a** [comment]
Hey,

The logging parameters  are usually found in `def setup_model(self):` and look like `tf.summary.[type]([name], [value])`, if you comment everything except the scalars that should reduce by an order of magnitude the logging size.

**Comment by SerialIterator** [comment]
Awesome!!! No more crashing and logs are about 100x smaller. Good job guys!

**Comment by araffin** [comment]
Closed by #185
